### PR TITLE
Fixed #30614 -- Added constraint name validation to system checks.

### PR DESCRIPTION
**Pull request by cansarigol**
Ticket: [#30614](https://code.djangoproject.com/ticket/30614)

### CODE REVIEW

File: django/db/models/base.py [hunk 1]
```diff
@@ -1814,6 +1814,27 @@ def _check_long_column_names(cls):
     @classmethod
     def _check_constraints(cls):
         errors = []
+        for constraint in cls._meta.constraints:
```
Comment (cansarigol) [inline django/db/models/base.py]: this part would be a function for constraints and indexes?

Comment (felixxm) [reply django/db/models/base.py]: Yes we can add `_check_object_identifiers()` and share the same errors:
```
* **models.E033**: The index/constraint name ``<index>`` cannot start with an underscore or a number.
* **models.E034**: The index/constraint name ``<index>`` cannot be longer than ``<max_length>`` characters.
```

File: tests/invalid_models_tests/test_models.py [hunk 2]
```diff
@@ -1191,3 +1191,36 @@ class Meta:
         )
         expected = [] if connection.features.supports_table_check_constraints else [warn, warn]
         self.assertCountEqual(errors, expected)
+
+    def test_name_constraints(self):
```
Comment (pope1ni) [inline tests/invalid_models_tests/test_models.py]: Add the `@skipUnlessDBFeature('supports_table_check_constraints')` decorator.

File: tests/invalid_models_tests/test_models.py [hunk 3]
```diff
@@ -1191,3 +1191,36 @@ class Meta:
...
+        self.assertEqual(Model.check(), [
+            Error(
+                "The constraint name '%sconstraint_name' cannot start with an "
+                "underscore or a number." % prefix,
+                obj=Model,
+                id='models.E033',
+            ) for prefix in ('_', '5')
+        ])
+
+    def test_max_name_length(self):
```
Comment (pope1ni) [inline tests/invalid_models_tests/test_models.py]: Add the `@skipUnlessDBFeature('supports_table_check_constraints')` decorator.

**Comment by cansarigol** [comment]
Thanks @pope1ni I missed it I will add asap.

**Comment by felixxm** [comment]
Thanks for this patch :+1: and I'm really sorry but it is backward incompatible. I don't think that we should add this restriction.

Closing per ticket.

**Pull request closed without merging**